api/desecapi/tests/test_rrsets.py

```diff
@@ -268,8 +269,9 @@ def test_delete_my_rr_sets_with_delete(self):
     def test_delete_other_rr_sets(self):
         for subname in self.SUBNAMES:
             # Try PATCH empty
-            response = self.client.patch_rr_set(self.other_rr_set_domain.name, subname=subname, type_='A', records=[])
-            self.assertStatus(response, status.HTTP_204_NO_CONTENT)
+            # FIXME consider removing this functionality
+            # response = self.client.patch_rr_set(self.other_rr_set_domain.name, subname=subname, type_='A', records=[])
```

Comment:
This is for `PATCH`ing RRsets on someone else's domain, right? I guess we should give `403` or `404`.

Reply:
The rationale why we chose 204 here is that we didn't wanted to disclose existence of this object, and deletion of non-existing objects results in 204.

The comment however refers to the functionality that RRsets can be deleted by setting the record array to `[]`. Outside the bulk scope, that is somewhat weird and, as far as I can tell right now, not straight-forward to implement in the serializer layer.

Reply:
> The rationale why we chose 204 here is that we didn't wanted to disclose existence of this object, and deletion of non-existing objects results in 204.

I remember the discussion, and I tend to agree that the status should be 204 if the non-existing object is the endpoint "leaf", e.g. `DELETE /domains/desec.io/rrsets/@/A/`, and the `A` RRset does not exist. However, if the domain already does not exist, I would expect that `DELETE /domains/desec.io/` gives 204, and requests with any other path appended at the end give `404`.

I understand the non-disclosure argument, but our domains are public anyways.

Just my 2ct... I think that in the end, it does not matter how we do it.

> The comment however refers to the functionality that RRsets can be deleted by setting the record array to `[]`. Outside the bulk scope, that is somewhat weird and, as far as I can tell right now, not straight-forward to implement in the serializer layer.

I actually consider this functionality quite important, for atomic updates (remove `A` and add `CNAME` at the same time, or doing atomic updates to the PSL, or ...). Let's try to figure out how to best to it

Reply:
Which code we're using here should be discussed in a separate issue. I'm only considering to remove the quirky `[]` behavior for the non-bulk case.